File: sdk/nodejs/src/resources/network_services.js
"use strict";

const { BaseResource } = require("./base");

class NetworkServicesResource extends BaseResource {
  async getIps(serviceId) {
    return this._http.get("/api/service/{id}/ip");
  }

  async getRdns(serviceId) {
    return this._http.get("/api/service/{id}/rdns");
  }

  async setRdns(serviceId, data) {
    return this._http.post("/api/service/{id}/rdns", data);
  }

}

module.exports = { NetworkServicesResource };

File: sdk/java/src/main/java/vn/hitechcloud/sdk/resources/NetworkServices.java
package vn.hitechcloud.sdk.resources;

import java.util.Map;
import vn.hitechcloud.sdk.HttpClient;

public class NetworkServices extends BaseResource {

    public NetworkServices(HttpClient http) {
        super(http);
    }

    /** Get IPs */
    public Map<String, Object> ips(int id) throws Exception {
        return http.get(String.format("/api/service/%id/ip", id));
    }

    /** Get rDNS */
    public Map<String, Object> rdns(int id) throws Exception {
        return http.get(String.format("/api/service/%id/rdns", id));
    }

    /** Set rDNS */
    public Map<String, Object> setRdns(int id, Map<String, Object> data) throws Exception {
        return http.post(String.format("/api/service/%id/rdns", id), data);
    }

}
